realm/realm-library/src/main/java/io/realm/log/RealmLog.java

```diff
@@ -16,83 +16,154 @@
+     * @deprecated use {@link #add(RealmLogger)} instead.
+     */
+    public static void add(Logger logger) {
+        synchronized (LoggerAdaptor.class) {
+            add(new LoggerAdaptor(logger));
         }
     }
 
-    private static void setMinimumNativeDebugLevel(int nativeDebugLevel) {
-        minimumNativeLogLevel = nativeDebugLevel;
-        Util.setDebugLevel(nativeDebugLevel); // Log level for Realm Core
+    /**
+     * Sets the current {@link LogLevel}.
+     *
+     * @param level see {@link LogLevel}.
+     */
+    public static void setLevel(int level) {
+        nativeSetLogLevel(level);
+    }
+
+    /**
+     * Get the current {@link LogLevel}.
+     *
+     * @return the current {@link LogLevel}.
+     */
+    public static int getLevel() {
+        return nativeGetLogLevel();
     }
 
     /**
      * Removes the given logger if it is currently added.
      *
      * @return {@code true} if the logger was removed, {@code false} otherwise.
      */
-    public static boolean remove(Logger logger) {
+    public static boolean remove(RealmLogger logger) {
         if (logger == null) {
             throw new IllegalArgumentException("A non-null logger has to be provided");
         }
-        synchronized (LOGGERS) {
-            LOGGERS.remove(logger);
-            int newMinLevel = Integer.MAX_VALUE;
-            for (int i = 0; i < LOGGERS.size(); i++) {
-                int logMin = LOGGERS.get(i).getMinimumNativeDebugLevel();
-                if (logMin < newMinLevel) {
-                    newMinLevel = logMin;
-                }
+        nativeRemoveLogger(logger);
+        return true;
+    }
+
+    /**
+     * Removes the given logger if it is currently added.
+     *
+     * @return {@code true} if the logger was removed, {@code false} otherwise.
+     * @deprecated use {@link #remove(RealmLogger)} instead.
+     */
+    public static boolean remove(Logger logger) {
+        synchronized (LoggerAdaptor.class) {
+            if (logger == null) {
+                throw new IllegalArgumentException("A non-null logger has to be provided");
+            }
+            RealmLogger adaptor = LoggerAdaptor.removeLogger(logger);
+            if (adaptor != null) {
+                nativeRemoveLogger(adaptor);
             }
-            setMinimumNativeDebugLevel(newMinLevel);
-            loggersAsArray = LOGGERS.toArray(new Logger[LOGGERS.size()]);
         }
         return true;
     }
 
     /**
-     * Remove all loggers.
+     * Removes all loggers except the default logger. The default native logger will be removed as well. Use
+     * {@link #registerDefaultLogger()} to add it back.
      */
     public static void clear() {
-        synchronized (LOGGERS) {
-            LOGGERS.clear();
-            setMinimumNativeDebugLevel(Integer.MAX_VALUE);
-            loggersAsArray = NO_LOGGERS;
-        }
+        nativeClearLoggers();
+    }
+
+    /**
+     * Adds default native logger if it has been removed before.
```

Comment:
Also specify what happens if this is called twice? Is the default logger added twice or only once?